include/spdk/env.h

```diff
@@ -323,6 +335,546 @@ int spdk_pci_addr_fmt(char *bdf, size_t sz, const struct spdk_pci_addr *addr);
  */
 void *spdk_call_unaffinitized(void *cb(void *arg), void *arg);
 
+struct spdk_ring;
+
+#define RING_F_SP_ENQ 0x0001 /**< The default enqueue is "single-producer". */
+#define RING_F_SC_DEQ 0x0002 /**< The default dequeue is "single-consumer". */
+
+/**
+ * Enqueue several objects on the ring (multi-producers safe).
+ *
+ * This function uses a "compare and set" instruction to move the
+ * producer index atomically.
+ *
+ * \param sr
+ *   A pointer to the spdk_ring structure.
+ * \param obj_table
+ *   A pointer to a table of void * pointers (objects).
+ * \param n
+ *   The number of objects to add in the ring from the obj_table.
+ * \return
+ *   - 0: Success; objects enqueue.
+ *   - -EDQUOT: Quota exceeded. The objects have been enqueued, but the
+ *     high water mark is exceeded.
+ *   - -ENOBUFS: Not enough room in the ring to enqueue, no object is enqueued.
+ */
+int
+spdk_ring_mp_enqueue_bulk(struct spdk_ring *sr, void *const *obj_table,
```

Comment:
Function declarations should not split the line after the return type (only function definitions should do that).

There are some places where we aren't consistent on this, but we should fix those and avoid adding new ones.